Makefile

```diff
@@ -1,262 +1,222 @@
-# author: Sylvain Anderegg
-
-VERSION := $(shell uname -a)
-
-# SAN: this is a hack so that docker-compose works in the linux virtual environment under Windows
-WINDOWS_MODE=OFF
-ifneq (,$(findstring Microsoft,$(VERSION)))
-$(info    detected WSL)
-export DOCKER_COMPOSE=docker-compose
-export DOCKER=docker
-# SAN: Windows does not have these things defined... but they are needed to execute a local swarm
-WINDOWS_MODE=ON
-else ifeq ($(OS), Windows_NT)
-$(info    detected Powershell/CMD)
-export DOCKER_COMPOSE=docker-compose.exe
-export DOCKER=docker.exe
-WINDOWS_MODE=ON
-else ifneq (,$(findstring Darwin,$(VERSION)))
-$(info    detected OSX)
-SHELL = /bin/bash
-export DOCKER_COMPOSE=docker-compose
-export DOCKER=docker
+# osparc-simcore general makefile
+#
+# TODO: make fully windows-friendly (e.g. some tools to install or replace e.g. date, ...  )
+#
+# Recommended
+#   GNU make version 4.2
+#
+# by sanderegg, pcrespov
+#
+PREDEFINED_VARIABLES := $(.VARIABLES)
+
+# TOOLS --------------------------------------
+
+# Operating system
+ifeq ($(filter Windows_NT,$(OS)),)
+IS_LINUX:= $(filter Linux,$(shell uname))
+IS_OSX  := $(filter Darwin,$(shell uname))
 else
-$(info    detected native linux)
-SHELL = /bin/bash
-export DOCKER_COMPOSE=docker-compose
-export DOCKER=docker
+IS_WSL  := $(filter Microsoft,$(shell uname))
 endif
+IS_WIN  := $(if $(or IS_LINUX,IS_OSX,IS_WSL),,$(OS))
+
+$(info Detected OS : $(IS_LINUX) $(IS_OSX) $(IS_WSL) $(IS_WIN))
+
+
+# Makefile's shell
+SHELL = $(if $(IS_WIN),cmd.exe,/bin/bash)
+
+export MKDIR  = $(if $(IS_WIN),md,mkdir -p)
+export RM     = $(if $(IS_WIN),del /Q,rm)
+export MKTEMP = $(if $(IS_WIN),\
+	set "tmpdir=%temp%\mktemp~%RANDOM%" && md %tmpdir% && echo %tmpdir%,\
+	mktemp)
```

Comment:
does that work under wsl? as far as I know the /tmp directory is not accessible if mounted in docker through wsl (Windows-WSL with docker cannot mount anything located inside the wsl virtual machine, only folders accessible by Windows Explorer). Maybe this will work in WSL2.